### PR TITLE
bugfix - refactor loading of UI data

### PR DESCRIPTION
**Pull request by sjacobs1**
##Description
refactored functionality for loading UI data 

##Screenshots
n/a

##Checklist
- [x] The code follows the project's coding standards.
- [ ] 100% Unit test coverage for Tested View Models
- [x] All warnings Resolved.

### CODE REVIEW

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift [hunk 1]
```diff
@@ -7,6 +7,10 @@
 
 import UIKit
 
+protocol MovieDetailsView: AnyObject {
+    func updateUI()
+}
+
```
Comment (sjacobs1) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift:13]: implemented protocol for updating ui

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift [hunk 2]
```diff
@@ -61,6 +59,7 @@ class MovieDetailsViewModel {
             switch result {
             case .success(let details):
                 self?.movieDetails = details
+                self?.view?.updateUI()
             case .failure(let error):
                 print(error)
```
Comment (sjacobs1) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:64]: update ui no longer called in viewDidLoad

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift [hunk 3]
```diff
@@ -61,6 +59,7 @@ class MovieDetailsViewModel {
...
             }
```
Comment (sjacobs1) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:66]: print statements need to be removed. I will implement error management in error management pr

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift [hunk 4]
```diff
@@ -61,6 +59,7 @@ class MovieDetailsViewModel {
             switch result {
             case .success(let details):
                 self?.movieDetails = details
+                self?.view?.updateUI()
             case .failure(let error):
```
Comment (sjacobs1) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:63]: update ui no longer called in viewDidLoad

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift [hunk 5]
```diff
@@ -7,6 +7,10 @@
 
 import UIKit
 
+protocol MovieDetailsView: AnyObject {
+    func updateUI()
+}
```
Comment (simekanidev) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift]: Please move this to view model?

Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift]: will do

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift [hunk 6]
```diff
@@ -30,22 +30,17 @@ class MovieDetailsViewController: UIViewController {
     // MARK: - Functions
     override func viewDidLoad() {
         super.viewDidLoad()
-        updateUI()
         movieDetailsViewModel.fetchMovieDetails()
-        setupViewModelObserver()
+        movieDetailsViewModel.view = self
```
Comment (kieran-woodrow) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift]: this is very strange? why is this needed..

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift [hunk 7]
```diff
@@ -7,16 +7,20 @@
 
 import Foundation
```
Comment (kieran-woodrow) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:8]: why do your files have a dash in then for the name?
should just be MovieDetailsViewModel.swift
remove Get as well

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift [hunk 8]
```diff
@@ -7,16 +7,20 @@
...
     // MARK: - Variables
     private let movieDetailsRepository: MovieDetailsRepositoryType
     private var movieID = 0
-    var movieDetails: MovieDetails? {
-        didSet {
-            didUpdateDetails?()
-        }
-    }
+    var movieDetails: MovieDetails?
+    weak var view: MovieDetailsView?
```
Comment (kieran-woodrow) [inline MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift]: this is very very very strange. The viewModel should never ever be coupled to the view. Please change this. This can't be approved. It breaks architecture in every regard.
The view knows about the viewModel, not the other way round

**Review by kieran-woodrow — CHANGES_REQUESTED** [review]

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift (hunk 7 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift:8]: understood

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift (hunk 6 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewController.swift]: it was used as part of loading the UI, will address

File: MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift (hunk 8 above)
Comment (sjacobs1) [reply MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift]: will address

**Review by simekanidev — APPROVED** [review]

**Review by kieran-woodrow — APPROVED** [review]
